[deep semgrep] `semgrep --interfile` fails with wrong, generic error message

**Describe the bug**

A useless error message is produced when semgrep is invoked with the `--interfile` option and `semgrep-core-proprietary` fails to parse a rule due to an invalid pattern.

Since #5996 , the error message is:
```
Exception raised: `Error while running rules: semgrep-core exited successfully but produced unexpected output`
```
It used to be:
```
Exception raised: `Error while running rules: 0 bytes read on a total of 2 expected bytes`
```

* Replacing `semgrep-core-proprietary` by a copy of `semgrep-core` doesn't change a thing.
* Using `--pro` works fine. It's the error reporting with `--interfile` that's broken.
* `semgrep-core-proprietary`, when run alone, exits with code 2, unlike what the new error message claims (sorry about that - @mjambon)

**To Reproduce**

```
$ cat toto.yaml 
rules:
  - id: apex.injection.soql-injection
    severity: ERROR
    languages:
      - python
    message: foo
    pattern: "if True:"
```

This is the semgrep command that shows the unhelpful error message. Note that to reproduce the problem, you can copy the regular `semgrep-core` to the `bin/` folder where `semgrep-core-proprietary` is expected:
```
$ semgrep scan --config toto.yaml --interfile
Scanning 19 files.
!!!This is a proprietary extension of semgrep.!!!
!!!You must be logged in to access this extension!!!
Using Semgrep Pro installed in /home/martin/semgrep/cli/src/semgrep/bin/semgrep-core-proprietary
Semgrep Pro Version Info: (0.13.0-96-g526b2f0)
    0%|                                                              |0/19 tasks

Semgrep Pro crashed during execution (unknown reason).
This can sometimes happen because either Semgrep Pro or Semgrep is out of date.

Try updating your version of Semgrep Pro (`semgrep install-semgrep-pro`) or your version of Semgrep (`pip install semgrep/brew install semgrep`).
If both are up-to-date and the crash persists, please contact support to report an issue!
When reporting the issue, please re-run the semgrep command with the
`--debug` flag so as to print more details about what happened, if you can.

Exception raised: `Error while running rules: semgrep-core exited successfully but produced unexpected output`
                    
    0%|                                                              |0/19 tasks
```

**Expected behavior**

This the desired output that we get without the `--interfile` option:
```
$ sg scan --config toto.yaml
Scanning 19 files.
    0%|                                                              |0/19 tasks
[ERROR] Pattern parse error in rule apex.injection.soql-injection:
 Invalid pattern for Python:
--- pattern ---
if True:
--- end pattern ---
Pattern error: Parsing.Parse_error
```

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**

CLI, normal build on Linux

**Use case**

Allow any user of `--interfile` to see what's wrong when they write an invalid pattern (or possibly other kinds of errors) without going through the Semgrep Playground.

--- a/changelog.d/gh-7028.fixed
+++ b/changelog.d/gh-7028.fixed
@@ -1,1 +1,2 @@
-Fix: Semgrep Pro previously reported a crash for normal errors. This fixes that
+Fix: Semgrep Pro previously reported a crash for user errors such as
+invalid patterns. It will now give a good error message.

--- a/changelog.d/gh-7028.fixed
+++ b/changelog.d/gh-7028.fixed
@@ -1,0 +1,1 @@
+Fix: Semgrep Pro previously reported a crash for normal errors. This fixes that

--- a/cli/src/semgrep/core_runner.py
+++ b/cli/src/semgrep/core_runner.py
@@ -992,7 +992,11 @@ class CoreRunner:
             return self._run_rules_direct_to_semgrep_core_helper(
                 rules, target_manager, dump_command_for_core, engine
             )
+        except SemgrepError as e:
+            # Handle Semgrep errors normally
+            raise e
         except Exception as e:
+            # Unexpected error, output a warning that the engine might be out of date
             if engine.is_pro:
                 logger.error(
                     f"""

